var/spack/repos/builtin/packages/cairo/package.py

```diff
@@ -51,4 +51,6 @@ class Cairo(AutotoolsPackage):
     def configure_args(self):
         args = ["--disable-trace",  # can cause problems with libiberty
                 "--enable-tee"]
+        if self.spec.satisfies("~X"):
+            args.extend(["--disable-xlib", "--disable-xcb"])
```

Comment:
Two requests:

1. I prefer `'+X' in self.spec` over `self.spec.satisfies('+X')`. The latter is a little unpredictable (see #4135)
2. For the sake of symmetry, can you also explicitly `--enable-xlib` and `--enable-xcb` for `+X`?